doc/template_definition.md

```diff
@@ -262,9 +376,106 @@ String someMethod() {
 }
 ```
 
+## PatternBuilder parameters
+The `PatternBuilder` takes all the Template parameters mentioned in the chapters above
+and understands them as template parameters, when `PatternBuilder#configureTemplateParameters()`
+is called.
+```java
+Pattern t = PatternBuilder.create(...select template model...)
+	.configureTemplateParameters()
+	.build();
+```
+
+Next to the ways of parameter definitions mentioned above the `PatternBuilder`
+allows to define parameters like this:
+```java
+Pattern t = PatternBuilder.create(...select template model...)
+	.configureParameters(pb -> 
+		pb.parameter("firstParamName")
+			//...select which AST nodes are parameters...
+			//e.g. using parameter selector
+			.byType(Request.class)
+			//...modify behavior of parameters...
+			//e.g. using parameter modifier
+			.setMinOccurence(0);
```

Comment:
I did not understood this example: is there a template example attached?

Reply:
It is not really an example which would do something real ... it is just an example which shows where to use `template parameter selectors` and where to use  `template parameter modifiers`

Reply:
ok so here, you will create a map of parameters, where `firstParamName` is the value of the first optional variable of type `Request`? or should `firstParamName` be the name of the variable?

Reply:
"firstParamName" is the name of the template parameter and all the CtTypeReferences which points to `Request.class` will be substituted by that parameter.

Reply:
hmmm, so here the `setMinOccurrence` is only used to skip some values if needed?

For example, if the following method is matched:

```java

Request foo(Request bar) {
   Request result = ... 
   return result;
}
```

if I used setMinOccurence(0) it takes all `Request`: the CtReturn, `bar` and `result`; if I used `setMinOccurence(1)` it takes only `bar` and `result` etc?

Reply:
`setMinOccurence` combined with `byType` is bad example. I will search for a better one ... because this combination makes no sense together.
May be search for setMinOccurence in test cases of this PR. There are some running examples.

Reply:
```java
//a template model 
void method(String _x_) {
	zeroOneOrMoreStatements();
	System.out.println(_x_);
}

//a pattern definition
Pattern t = PatternBuilder.create(...select template model...)
	.configureParameters(pb -> 
		pb.parameter("firstParamName")
			//...select which AST nodes are parameters...
			//e.g. using parameter selector
			.bySimpleName("zeroOneOrMoreStatements")
			//...modify behavior of parameters...
			//e.g. using parameter modifier
			.setMinOccurence(0);
			
		//... you can define as many parameters as you need...
		
		pb.parameter("lastParamName").byVariable("_x_");
	)
	.build();
```
next commit will contain this example. Here the the invocation of `zeroOneOrMoreStatements();` represents parameter "firstParamName", whose value will match with 0, 1 or more statements.